src/org/zaproxy/zap/extension/pscanrulesAlpha/XDebugTokenScanner.java

```diff
@@ -0,0 +1,170 @@
+ * 
+ * @author kingthorin+owaspzap@gmail.com
+ */
+public class XDebugTokenScanner extends PluginPassiveScanner {
+
+	private static final String MESSAGE_PREFIX = "pscanalpha.xdebugtoken.";
+	private static final int PLUGIN_ID = 10056;
+
+	private PassiveScanThread parent = null;
+	private static final Logger LOGGER = Logger.getLogger(XDebugTokenScanner.class);
+
+	private static final String X_DEBUG_TOKEN_HEADER = "X-Debug-Token";
+	private static final String X_DEBUG_TOKEN_LINK_HEADER = "X-Debug-Token-Link";
+	private static final String PROFILER_PORTION = "_profiler"; // http://example.com/_profiler/c6bd78
+
+	@Override
+	public void setParent(PassiveScanThread parent) {
+		this.parent = parent;
+	}
+
+	@Override
+	public void scanHttpRequestSend(HttpMessage msg, int id) {
+		// Only checking the response for this plugin
+	}
+
+	@Override
+	public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
+		long start = System.currentTimeMillis();
+
+		//Check "Link" variant first as it's of greater concern/convenience.
+		Vector<String> xdtlHeaders = msg.getResponseHeader().getHeaders(X_DEBUG_TOKEN_LINK_HEADER);
+		if (xdtlHeaders != null) { // Header Found
+			for (String xdtl : xdtlHeaders) {
+				parent.raiseAlert(id, getAlert(msg, X_DEBUG_TOKEN_LINK_HEADER));
+				return; // No need to continue
+			}
+		}
+		// Check non-Link variant
+		Vector<String> xdtHeaders = msg.getResponseHeader().getHeaders(X_DEBUG_TOKEN_HEADER);
+		if (xdtHeaders != null) { // Header Found
+			for (String xdt : xdtHeaders) {
+				parent.raiseAlert(id, getAlert(msg, X_DEBUG_TOKEN_HEADER));
+			}
+		}
+		
+		if (LOGGER.isDebugEnabled()) {
+			LOGGER.debug("\tScan of record " + id + " took " + (System.currentTimeMillis() - start) + " ms");
+		}
+	}
+	
+	private Alert getAlert(HttpMessage msg, String evidence) {
+		Alert alert = new Alert(getPluginId(), Alert.RISK_LOW, Alert.CONFIDENCE_HIGH, // PluginID, Risk, Reliability
+				getName());
+		alert.setDetail(getDescription(), // Description
+				msg.getRequestHeader().getURI().toString(), // URI
+				"", // Param
+				"", // Attack
+				getOtherInfo(getProfilerLink(msg, evidence)), // Other info
+				getSolution(), // Solution
+				getReference(), // References
+				evidence, // Evidence - Return the Server Header info
+				200, // CWE-200: Information Exposure
+				13, // WASC Id
+				msg); // HttpMessage
+		return alert;
+	}
+	
+	private String getProfilerLink(HttpMessage msg, String headerValue) {
+		URI assembledURI = null;
+		try {
+			assembledURI = new URI(headerValue, false);
+		} catch (URIException e) {
+			if (LOGGER.isDebugEnabled()) {
+				LOGGER.debug("Failed to create URI", e);
+			}
+			return "";
+		}
+		if (assembledURI.isAbsoluteURI()) { // Full URL: http://example.com/_profiler/3d02cb
+			return assembledURI.toString();
+		} else if (headerValue.toLowerCase().contains(PROFILER_PORTION)) { // Full Path: /_profiler/3d02cb 
+			URI currentURI = msg.getRequestHeader().getURI();

```

Comment:
It could be extracted a method that accepts the current URI and a path, it would be used by both ifs (full path/token only).

Worth adding tests for these cases too.